src/hub.js

```diff
@@ -317,8 +319,19 @@ async function updateEnvironmentForHub(hub) {
     THREE.Cache.clear();
 
     const onLoadingEnvironmentReady = () => {
-      environmentEl.setAttribute("gltf-model-plus", { src: sceneUrl });
       environmentEl.removeEventListener("model-loaded", onLoadingEnvironmentReady);
+
+      if (scene.is("entered")) {
+        // We've already entered, so move to new spawn point
+        const spawnWhenEnvironmentReady = () => {
+          environmentEl.removeEventListener("model-loaded", spawnWhenEnvironmentReady);
```

Comment:
You can use `{once: true}` on the `addEventListener` instead of dealing with this cleanup. That also lets you just inline the function since you dont need a ref to it anymore.

Reply:
nice trick!